spark/src/test/scala/org/apache/comet/exec/CometAggregateSuite.scala

```diff
@@ -38,6 +38,25 @@ import org.apache.comet.CometSparkSessionExtensions.isSpark34Plus
  * Test suite dedicated to Comet native aggregate operator
  */
 class CometAggregateSuite extends CometTestBase with AdaptiveSparkPlanHelper {
+  import testImplicits._
+
+  test("Final aggregation should not bind to the input of partial aggregation") {
```

Comment:
hmm does this test the issue in this PR? seems it passes in the `main` branch without the fix.

Reply:
After adding the bound index check, it fails. It passes now because we don't have the check now. As I mentioned in the description, the bound is incorrect but because we don't check it, it is not exposed.

Reply:
Btw, we need to add the bound check as required for SortMergeJoin work. In that work, it checks joining keys bindings internally in DataFusion. Which checks both binding index and binding column name. So we need to bind column reference to input schema.